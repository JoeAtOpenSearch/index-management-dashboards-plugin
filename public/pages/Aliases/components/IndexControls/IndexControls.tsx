/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { Component } from "react";
import { EuiFlexGroup, EuiFlexItem, EuiSearchBar } from "@elastic/eui";

interface SearchControlsProps {
  search: string;
  onSearchChange: (args: any) => void;
}

export default class SearchControls extends Component<SearchControlsProps, {}> {
  render() {
    const { search, onSearchChange } = this.props;

    const schema = {
      strict: true,
      fields: {
        indices: {
          type: "string",
        },
      },
    };

    return (
      <EuiFlexGroup style={{ padding: "0px 5px" }} alignItems="center">
        <EuiFlexItem>
          <EuiSearchBar query={search} onChange={onSearchChange} />
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  }
}
