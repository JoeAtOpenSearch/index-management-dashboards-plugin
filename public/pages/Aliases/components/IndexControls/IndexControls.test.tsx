/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
// @ts-ignore
import userEvent from "@testing-library/user-event";
import IndexControls from "./IndexControls";

describe("<IndexControls /> spec", () => {
  it("renders the component", async () => {
    const { container } = render(<IndexControls value={{ search: "testing", status: "1" }} onSearchChange={() => {}} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
