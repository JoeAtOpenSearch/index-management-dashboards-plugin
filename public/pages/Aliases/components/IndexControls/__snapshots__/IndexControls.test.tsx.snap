// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexControls /> spec renders the component 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
  style="padding: 0px 5px;"
>
  <div
    class="euiFlexItem"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
    >
      <div
        class="euiFlexItem euiSearchBar__searchHolder"
      >
        <div
          class="euiFormControlLayout euiFormControlLayout--fullWidth"
        >
          <div
            class="euiFormControlLayout__childrenWrapper"
          >
            <input
              aria-label="This is a search bar. After typing your query, hit enter to filter the results lower in the page."
              class="euiFieldSearch euiFieldSearch--fullWidth euiFieldSearch-isClearable"
              placeholder="Search..."
              type="search"
              value="testing"
            />
            <div
              class="euiFormControlLayoutIcons"
            >
              <span
                class="euiFormControlLayoutCustomIcon"
              >
                EuiIconMock
              </span>
            </div>
            <div
              class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Clear input"
                class="euiFormControlLayoutClearButton"
                type="button"
              >
                EuiIconMock
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
